### PR TITLE
HPCC-14059 Global Thor debug options not read due to getOpt* bug

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith jake.smith@lexisnexis.com

### CODE REVIEW

File: thorlcr/graph/thgraph.cpp [hunk 1]
```diff
@@ -2516,7 +2516,7 @@ void CJobBase::init()
     thorAllocator.setown(createThorAllocator(((memsize_t)globalMemorySize)*0x100000, memorySpillAt, *logctx, crcChecking, usePackedAllocator));
 
     unsigned defaultMemMB = globalMemorySize*3/4;
-    unsigned largeMemSize = getOptUInt("@largeMemSize", defaultMemMB);
+    unsigned largeMemSize = getOptUInt("largeMemSize", defaultMemMB);
```
Comment (jakesmith) [inline thorlcr/graph/thgraph.cpp]: Not directly related, but correction for Debug/ , caused this to fail (i.e. was looking for @@largeMemSize)
largeMemSize is legacy anyway , I will remove related code in another PR.

**Comment by jakesmith** [comment]
@richardkchapman - please review

File: thorlcr/graph/thgraphslave.cpp [hunk 2]
```diff
@@ -1028,7 +1028,7 @@ class CThorCodeContextSlave : public CThorCodeContextBase, implements IEngineCon
     virtual bool allowDaliAccess() const
     {
         // NB. includes access to foreign Dalis.
-        return globals->getPropBool("Debug/@slaveDaliClient") || job.getWorkUnitValueBool("slaveDaliClient", false);
```
Comment (richardkchapman) [inline thorlcr/graph/thgraphslave.cpp]: There is at least one other instance of code that calls getWorkUnitValueBool("slaveDaliClient") that should also be changed?

**Comment by richardkchapman** [comment]
@jakesmith One minor comment

**Comment by jakesmith** [comment]
No, they are slightly different, they are checking if global option is off and workunit option is on, so as to switch on/off dali client mode for the job only if global was off.

**Comment by jakesmith** [comment]
@richardkchapman

**Pull request merged**